python/src/energuide/snippets.py

```diff
@@ -178,6 +180,10 @@ def snip_house(house: etree._Element) -> HouseSnippet:
     ventilation = house.xpath('Ventilation/WholeHouseVentilatorList/Hrv')
     ventilation_strings = [etree.tostring(hrv, encoding='unicode') for hrv in ventilation]
 
+    water_heating = house.xpath('Components/HotWater/*[self::Primary or self::Secondary]')
```

Comment:
Look at you with the conditionals. I would have thought `[Primary or Secondary]` would have done it, but I'll trust you on this.

Reply:
Better to do this, or just `'Components/HotWater'`?

Reply:
If we do `Components/HotWater` then it gets a little annoying since it would make WaterHeating.from_data the first (and only) extracted datatype from_data method that returns a list of objects, instead of a singleton.

Not the end of the world of course, or even that big a deal, it just felt cleaner to me to keep that interface consistent.

Reply:
I'd rather keep the extractor dumb.